spiffe-helper.go

```diff
@@ -209,6 +219,28 @@ func (s *sidecar) writeKey(file string, data []byte) error {
 	return ioutil.WriteFile(file, pem.EncodeToMemory(b), keyFileMode)
 }
 
+// writeKey takes a private key as a slice of bytes,
+// formats as PEM, and appends it to file
+func (s *sidecar) appendKey(file string, data []byte) error {
+	b := &pem.Block{
+		Type:  "EC PRIVATE KEY",
+		Bytes: data,
+	}
+
+	f, err := os.OpenFile(file, os.O_WRONLY|os.O_CREATE|os.O_APPEND, keyFileMode)
+	if err != nil {
+		return err
+	}
+
+	_, err = f.Write(pem.EncodeToMemory(b))
+	if err != nil {
+		return err
```

Comment:
err = f.Close() is needed before returning here.

Reply:
`writeKey` and `appendKey` could be refactored to avoid some duplicated code

Reply:
refactored writeKey implementation to accomodate append mode.